doc/contribute.rst

```diff
@@ -0,0 +1,258 @@
+
+        $ git checkout -b my-feature
+
+   and start making changes. 
+
+.. note::
+    Never work in the ``master`` branch!
+
+4) Work on this copy on your computer using Git to do the version
+   control. When you're done editing, do::
+
+        $ git add modified_files
+        $ git commit
+
+   to record your changes in Git, then push them to GitHub with::
+
+        $ git push -u origin my-feature
+
+Finally, go to the web page of your fork of the ramp-board repo,
+and click 'Pull request' to send your changes to the maintainers for
+review. This will send an email to the committers.
+
+If any of the above seems like magic to you, then look up `Git documentation`_ 
+on the web.
+
+.. _Git documentation: https://git-scm.com/documentation
+
+
+Contributing Pull Requests
+==========================
+
+It is recommended to check that your contribution complies with the
+following rules before submitting a pull request:
+
+-  Follow the coding-guidelines_ as for scikit-learn.
+
+-  When applicable, use the validation tools and other code in the
+   `ramp_utils` subpackage.
+
+-  If your pull request addresses an issue, please use the title to describe
+   the issue and mention the issue number in the pull request description to
+   ensure a link is created to the original issue.
+
+-  All public methods should have informative docstrings with sample
+   usage presented as doctests when appropriate.
+
+-  Please prefix the title of your pull request with `[MRG]` if the
+   contribution is complete and should be subjected to a detailed review.
+   Incomplete contributions should be prefixed `[WIP]` to indicate a work
+   in progress (and changed to `[MRG]` when it matures). WIPs may be useful
+   to: indicate you are working on something to avoid duplicated work,
+   request broad review of functionality or API, or seek collaborators.
+   WIPs often benefit from the inclusion of a `task list`_ in the PR description.
+
+-  All other tests pass when everything is rebuilt from scratch. On
+   Unix-like systems, check with (from the toplevel source folder)::
+
+        $ make
+
+-  Documentation and high-coverage tests are necessary for enhancements
+   to be accepted.
+
+-  At least one paragraph of narrative documentation with links to
+   references in the literature (with PDF links when possible) and
+   the example.
+
+.. _coding-guidelines: http://scikit-learn.org/dev/developers/contributing.html#coding-guidelines
+.. _task list: https://github.com/blog/1375-task-lists-in-gfm-issues-pulls-comments
+
+You can also check for common programming errors with the following
+tools:
+
+-  Code with good unittest coverage (at least 80%), check with::
+
+        $ pip install pytest pytest-cov
+        $ pytest -vsl .
+
+-  No pyflakes warnings, check with::
+
+        $ pip install pyflakes
+        $ pyflakes path/to/module.py
+
+-  No PEP8 warnings, check with::
+
+        $ pip install pep8
+        $ pep8 path/to/module.py
```

Comment:
I would recomment flake8 that does both. And I would not recomment autopep8 as it can do horrors.

it's just a nitpick. Besides looks great !